### ISSUE TITLE
labels concatenated with features?

### ISSUE THREAD

**Issue by alejandrovr** [qq456cvb/PRIN#4]
Hi,

First of all, thanks for making the code public!

I have been looking at the code and I don't quite understand why the variable cat_onehot is being concatenated to the features of the point cloud when doing the forward pass.

```
#in model.py
def forward(self, x, target_index, cat_onehot)
    ....
    prediction = torch.cat([features, cat_onehot[:, None, :].repeat(1, features.size(1), 1)], dim=2)

#in train.py, train_step:
 _, prediction = model(data, target_index, cat_onehot)
```
Maybe I misunderstood something, could you explain why are you doint that cat operation?

Best and thanks again,

Alejandro.

**Comment by qq456cvb** [comment]
When doing segmention, one could either train a separate network for each category or a universal network for all categories (16 categories for shapenet) like we did here. If a universal network is used, then a category label is known but segmentation label is unknown (for each category there several segmentions), you can find a similar operation here in PointNet https://github.com/charlesq34/pointnet/blob/master/part_seg/pointnet_part_seg.py#L118

**Comment by alejandrovr** [comment]
That makes perfect sense. Thanks for the quick reply!
